framework/execution_framework/plugins/org.eclipse.gemoc.executionframework.extensions.sirius/src/org/eclipse/gemoc/executionframework/extensions/sirius/modelloader/DefaultModelLoader.java

```diff
@@ -304,76 +304,77 @@ protected void doExecute() {
 		for (DView view : session.getSelectedViews()) {
 			for (DRepresentationDescriptor repDescriptor : view.getOwnedRepresentationDescriptors()) {
 				DRepresentation representation = repDescriptor.getRepresentation();
-
-				final DSemanticDiagram diagram = (DSemanticDiagram) representation;
-				openEditorSubMonitor.subTask(diagram.getName());
-				final List<EObject> elements = new ArrayList<EObject>();
-				elements.add(diagram);
-
-				final IEditorPart editorPart = DialectUIManager.INSTANCE.openEditor(session, representation,
-						openEditorSubMonitor.newChild(1));
-				ToolManagement toolManagement = DiagramPlugin.getDefault().getToolManagement(diagram);
-				toolManagement.addToolFilter(new ToolFilter() {
+				if(representation instanceof DSemanticDiagram) {
```

Comment:
This line is in fact almost the only real change introduced in the PR, correct?

Reply:
Right for item 1 of the description
for item 2 see my second comment